grakn/service/Session/TransactionService.py

```diff
@@ -165,7 +165,12 @@ def send(self, request):
         return response
 
     def close(self):
-        with self._queue.mutex: # probably don't even need the mutex
-            self._queue.queue.clear()
-        self._queue.put(None)
-        self._closed = True
+        if not self._closed:
+            with self._queue.mutex: # probably don't even need the mutex
+                self._queue.queue.clear()
+            self._queue.put(None)
+            self._closed = True
+            # force exhaust the iterator so `onCompleted()` is called on the server
+            # note: next() was throwing an error sometimes, but would be preferable to a loop
+            for empty in self._response_iterator:
```

Comment:
@vmax maybe you have an idea on how to do this better. `next(...)` was not working I think!